### PR TITLE
Completely drop Django<1.11 and Python<3.5 support

### LINKED ISSUE #274: Compatibility issue with Django 3.0 due to django.utils.six
Django 3.0 scheduled for December removed Python 2 compatibility APIs used by django-post_office (django.utils.six, django.utils.encoding.python_2_unicode_compatible).

See https://docs.djangoproject.com/en/dev/releases/3.0/#removed-private-python-2-compatibility-apis

It would be wise to switch now to the direct use of six.

### PR DESCRIPTION
**Pull request by Mogost**
Fix #274

### CODE REVIEW

**Comment by selwin** [comment]
@Mogost we shouldn't require `six` as a dependency and drop Python 2 compatibility altogether, since Django no longer supports Python 2 anyway. Could you please update this PR?

**Comment by selwin** [comment]
The builds on Travis would need to pass before this PR can be merged :)

File: post_office/tests/test_backends.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def test_backend_attachments(self):
         message = EmailMessage('subject', 'body', 'from@example.com',
                                ['recipient@example.com'])
 
-        message.attach('attachment.txt', 'attachment content')
+        message.attach('attachment.txt', b'attachment content')
```
Comment (Mogost) [inline post_office/tests/test_backends.py]: @selwin Please check here carefully

**Comment by Mogost** [comment]
@selwin done! I doubt some of the changes, and I mentioned you to make sure that everything is fine there.

**Comment by Mogost** [comment]
Also build for djangomaster failed due to https://github.com/dmkoch/django-jsonfield/issues/225

File: post_office/tests/test_backends.py (hunk 1 above)
Comment (Mogost) [reply post_office/tests/test_backends.py]: All is ok
https://github.com/django/django/blob/e3c2fae4cd549a6aeefcf7b217ba4b173489e18c/tests/mail/tests.py#L491

File: post_office/tests/test_models.py [hunk 2]
```diff
@@ -296,7 +296,7 @@ def test_attachments_email_message(self):
                              [('test.txt', 'test file content', 'text/plain')])
         else:
             self.assertEqual(message.attachments,
-                             [('test.txt', b'test file content', None)])
+                             [('test.txt', 'test file content', None)])
```
[the remark this answers is not in the document]
Comment (Mogost) [reply post_office/tests/test_models.py]: https://github.com/django/django/blob/e3c2fae4cd549a6aeefcf7b217ba4b173489e18c/tests/mail/tests.py#L491

File: post_office/tests/test_models.py [hunk 3]
```diff
@@ -317,7 +317,7 @@ def test_attachments_email_message_with_mimetype(self):
                              [('test.txt', 'test file content', 'text/plain')])
         else:
             self.assertEqual(message.attachments,
-                             [('test.txt', b'test file content', 'text/plain')])
+                             [('test.txt', 'test file content', 'text/plain')])
```
[the remark this answers is not in the document]
Comment (Mogost) [reply post_office/tests/test_models.py]: https://github.com/django/django/blob/e3c2fae4cd549a6aeefcf7b217ba4b173489e18c/tests/mail/tests.py#L362

**Comment by Mogost** [comment]
@selwin I think that we need to merge these changes, but not yet release a new version. The problem #274  must be solved first

**Comment by Mogost** [comment]
@jrief can you review this changes?

**Comment by selwin** [comment]
Thanks!

**Pull request merged**